events/func.py

```diff
@@ -1,35 +1,115 @@
+import datetime
 import io
 import os
 import json
 import logging
 import requests
+import uuid
+
+import asyncio
+import fdk
+import oci
 
 from fdk import response
+from slack import WebClient
+from slack.errors import SlackApiError
 
 local_dev = os.environ.get('is_local')
+compartment_id = os.environ.get('compartment_id')
+slack_bot_token = os.environ.get('slack_bot_token')
+welcome_channel = os.environ.get('slack_welcome_channel')
+
+FORMAT = '%(asctime)s -- %(message)s'
+logging.basicConfig(format=FORMAT)
 logger = logging.getLogger()
 
+TABLE_NAME = "slack_events"
+
+slack_client = WebClient(token=slack_bot_token, run_async=True)
+
+
+def should_post_message(oci_client, event):
+    should_post = False
+
+    details = oci.nosql.models.QueryDetails(
+        compartment_id=compartment_id,
+        statement=f"SELECT * FROM {TABLE_NAME} e where e.event_type = 'member_joined_channel' AND e.data.user = '{event['user']}'"
+    )
+
+    try:
+        query_resp = oci_client.query(limit=1, query_details=details)
+        if len(query_resp.data.items) == 0:
+            should_post = True
+    except Exception as ex:
+        should_post = False
```

Comment:
This would already be false from line 32.

Don't forget that in python empty lists don't pass `if`, so `if len(x) == 0` is the same as `if not x`. But hey, if you charge for character then I say go for it.